corehq/apps/geospatial/routing_solvers/ortools.py

```diff
@@ -0,0 +1,166 @@
+        solution = routing.SolveWithParameters(search_parameters)
+
+        if not solution:
+            # Todo; custom exception type
+            raise Exception("No Solution Found")
+        # Print solution on console.
+        if print_routes:
+            self.print_solution(data, manager, routing, solution)
+        return self.user_to_case_map(routing, solution)
+
+    def print_solution(self, data, manager, routing, solution):
+        max_route_distance = 0
+        for vehicle_id in range(data['num_vehicles']):
+            index = routing.Start(vehicle_id)
+            plan_output = 'Route for vehicle {}:\n'.format(vehicle_id)
+            route_distance = 0
+            while not routing.IsEnd(index):
+                plan_output += ' {} -> '.format(manager.IndexToNode(index))
+                previous_index = index
+                index = solution.Value(routing.NextVar(index))
+                route_distance += routing.GetArcCostForVehicle(
+                    previous_index, index, vehicle_id)
+            plan_output += '{}\n'.format(manager.IndexToNode(index))
+            plan_output += 'Distance of the route: {}m\n'.format(route_distance)
+            print(plan_output)
+            max_route_distance = max(route_distance, max_route_distance)
+        print('Maximum of the route distances: {}m'.format(max_route_distance))
+
+    def user_to_case_map(self, routing, solution):
+        routes = []
+        for user_index in range(len(self.user_locations)):
+            index = routing.Start(user_index)
+            route = []
+            while not routing.IsEnd(index):
+                index = solution.Value(routing.NextVar(index))
+                route.append(index)
+            route.pop(-1)  # remove end location
+            routes.append(route)
+        ret = {}
+        for i, user in enumerate(self.user_locations):
+            ret[user['id']] = [
+                self.case_locations[case_index - len(self.user_locations)]['id']
+                for case_index in routes[i]
+            ]
+            routes[i]
+        return ret
+
+
+class ORToolsRoadNetworkSolver(ORToolsRadialDistanceSolver):
+    """
+    Solves user-case location assignment based on driving distance
+    """
+
+    def calculate_distance_matrix(self):
+        # Todo; support more than Mapbox limit by chunking
+        if len(self.user_locations + self.case_locations) > 25:
+            raise Exception("This is more than Mapbox matrix API limit")
+
+        coordinates = ';'.join([
+            f'{l["lon"]},{l["lat"]}'
+            for l in self.user_locations + self.case_locations]
+        )
+
+        url = f'https://api.mapbox.com/directions-matrix/v1/mapbox/driving/{coordinates}'
+        params = {
+            'annotations': 'distance',
+            'access_token': settings.MAPBOX_ACCESS_TOKEN
+        }
+
+        response = requests.get(url, params=params)
+        response.raise_for_status()
+        print(response.json()['distances'])
```

Comment:
Is this print statement meant to be here?

Reply:
Nope!